include/gz/rendering/base/BaseMaterial.hh

```diff
@@ -960,14 +1022,17 @@ namespace gz
       this->SetCastShadows(_material->CastShadows());
       this->SetReceiveShadows(_material->ReceiveShadows());
       this->SetReflectionEnabled(_material->ReflectionEnabled());
-      this->SetTexture(_material->Texture());
-      this->SetNormalMap(_material->NormalMap());
-      this->SetRoughnessMap(_material->RoughnessMap());
-      this->SetMetalnessMap(_material->MetalnessMap());
+      this->SetTexture(_material->Texture(), _material->TextureData());
+      this->SetNormalMap(_material->NormalMap(), _material->NormalMapData());
+      this->SetRoughnessMap(_material->RoughnessMap(),
+          _material->RoughnessMapData());
+      this->SetMetalnessMap(_material->MetalnessMap(),
+          _material->MetalnessMapData());
       this->SetRoughness(_material->Roughness());
       this->SetMetalness(_material->Metalness());
       this->SetEnvironmentMap(_material->EnvironmentMap());
-      this->SetEmissiveMap(_material->EmissiveMap());
+      this->SetEmissiveMap(_material->EmissiveMap(),
+          _material->EmissiveMapData());
```

Comment:
Not all the maps are implemented, specifically:

* Environment and Lightmap don't have a setter / getter for using gz::common::Image, this is intended (for now) since the primary use for this is using the GLTF format which can embed textures, however the [material specification](https://github.com/KhronosGroup/glTF/blob/main/specification/2.0/schema/material.schema.json) shows no mention of environment or light maps
* AmbientOcclusion _is_ in the specification but seems not to be supported at the `gz-rendering` level. Since from what I understand adding virtual member variables to Material / BaseMaterial breaks ABI we could add them and set them to noop if there is support in any rendering engine, to avoid having to postpone that to a next release.

Reply:
> Since from what I understand adding virtual member variables to Material / BaseMaterial breaks ABI we could add them and set them to noop if there is support in any rendering engine, to avoid having to postpone that to a next release.

+1, I think it's worth it adding them as noop for now to keep the API consistent and future-proof.

Reply:
Added Environment and Light maps https://github.com/gazebosim/gz-rendering/pull/642/commits/1fa4e88cde41700aed13bfe9f2346e4abf309e47, left is only a single TODO for binary environment maps but it's at the gz-common level since there is no use case for it (I couldn't find any format that supports it, and it can be added later without breaking API / ABI since `gz-common::Pbr` object doesn't rely on virtual functions).

From what I understand Ambient Occlusion and Light maps are quite similar and indeed `assimp` puts the glTF `ambientOcclusion` map inside the `LIGHTMAP` tag, hence that's where they are currently loaded.
There seems to be really no support anywhere in the stack for Ambient Occlusion maps in any of the parsers, sdformat, or rendering functions, so I didn't add those.